artifactory.py

```diff
@@ -1260,6 +1276,27 @@ def open(self, mode="r", buffering=-1, encoding=None, errors=None, newline=None)
 
         return self._accessor.open(self)
 
+    def download_large_file(self, out_file, chunk_size=1024, progress_func=print_download_progress, *args, **kwargs):
+        """
+        Downloads large file in chunks and prints progress
+        :param out_file: file path of output file
+        :param chunk_size: chunk size in bytes, recommend. eg 1024*1024 is 1Mb
+        :param progress_func: custom function to print output, otherwise uses default print_download_progress
+        :param args: *args to supply for your custom progress_func
```

Comment:
`*args, *kwargs` could be moved to `lambda` and to be mentioned in the README

```python
download_large_file("file.zip", progress_func=my_func, key=value)

# should be
download_large_file("file.zip", progress_func lamda offset. total: my_func(offset, total, key=value)
```

The library shouldn't know how to pass args to callbacks, just requires callback with a signature `[Int, Int] => None`

Reply:
@allburov 
if we provide such implementation then this will limit user functions for `progress_func`, won't it?

my idea was that with args and kwargs you can provide any kind of argument from "outside"

Reply:
here is example:
```python
path.download_large_file(output="tomcat2.tar.gz", chunk_size=1024*100, progress_func=None)
```

Reply:
>if we provide such implementation then this will limit user functions for progress_func, won't it?
>my idea was that with args and kwargs you can provide any kind of argument from "outside"

I meant that we can remove *args *kwargs and with python there are still several ways to fit into function signature:
1. `partial` - https://docs.python.org/3/library/functools.html#functools.partial
1. Create a `lambda` function
1. Create a wrapper function 

And we should use it

Reply:
@allburov 
I think using partial is quiet contr intuitive

how can we utilize lambda as argument in function definition and still pass args to it? I never used such functionality before

Reply:
may be I misunderstood the idea, but here is a use case I can think of.
Somebody would like to implement custom function with custom args.
```python
def custom_print(now, total, severity):
    if severity == 1:
         print("Important:", now, total) 
```

then they need to call:
```python
path.writeto(out, progress_func=custom_print, severity=1)
```

with current implementation I will pass kwargs and it will work. However how should we pass the data through partial or lambda ?
for me it looks like with such methods we can only hardcode some values, eg in default arguments or using globals, which is not nice

if you can provide me with sample I will be glad to implement it

Reply:
or do you mean to pass the burden on users and ask them to utilize like this?

```python
def custom_print(now, total, custom):
    print(now, total, custom)


def writeto(func):
    now = 1
    total = 2
    func(now, total)

writeto(func=lambda x, y: custom_print(x, y, custom="test"))
```

Reply:
>I think using partial is quiet contr intuitive

It's for OOP but not for the functional programming :)

> do you mean to pass the burden on users 
I wouldn't say it's burden. It's a good way to fit your function into a function signature. 
Just haven't seen the solution with `args, kwargs` that then will go to callback yet. 

You've got the idea - `lambda`, create a new function, `partial` what they like to reduce amount of arguments

Reply:
>path.writeto(out, progress_func=custom_print, severity=1)

When I'm reading this code (and haven't read the function signature yet) I expect that `severity` influences on 'wtireto`, not on `custom_print`. 

Another cons - there's no way to pass args kwargs to the function itself

Reply:
now I see, agree. Then let me just add example in documentation